### DISCUSSION TITLE
Incorrect dilect detection for SQLite on EL7 with SQLAlchemy 1.4.20

### DISCUSSION THREAD

**Discussion by jcpunk** [sqlalchemy/sqlalchemy#6773] [Usage Questions]
**Describe the bug**
SQLAlchemy 1.4.20 on EL7 expects the SQLite libraries to have support for the `WITH` operation.

**To Reproduce**
On an EL7 system with the system python 3.6.

sqlite-3.7.17-8.el7_7.1.x86_64

```shell
sudo yum -y install python3 python3-pip python3-wheel sqlite
python3 -m pip install --user sqlalchemy
```

```py
>>> subq = select(
...     func.count(address_table.c.id).label("count"),
...     address_table.c.user_id
... ).group_by(address_table.c.user_id).cte()

>>> stmt = select(
...    user_table.c.name,
...    user_table.c.fullname,
...    subq.c.count
... ).join_from(user_table, subq)
```

**Error**
sqlite3.OperationalError: near "WITH": syntax error

```
E       sqlite3.OperationalError: near "WITH": syntax error

context    = <sqlalchemy.dialects.sqlite.base.SQLiteExecutionContext object at 0x7fd5b078cf60>
cursor     = <sqlite3.Cursor object at 0x7fd5b07999d0>

```

**Versions.**
 - OS: RHEL7
 - Python: 3.6.8
 - SQLAlchemy: 1.4.20
 - Database: SQLite3
 - DBAPI: SQLite

**Additional context**
Support for the `WITH` option appears to be added in SQLite 3.8

**Have a nice day!**

**Comment by zzzeek** [comment]
> **Describe the bug**
> SQLAlchemy 1.4.20 on EL7 expects the SQLite libraries to have support for the `WITH` operation.

this is only if you wish to use the cte() construct.     As older versions of SQLite don't support this SQL construct, you can't use it when you are on those older backends.    There's no action SQLAlchemy can take here.
